data/tree_ring_data/build_tree_ring_file_19mar18.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python
+#Author: Craig Lage, UC Davis;
+#Date: 19-Mar-18
+
+# This program builds a file containing a set of tree ring models for the
+# detectors used in an imSim image simulation.  This file will be read into imSim
+# This guarantees that the tree ring model is the same each time imSim is run
+import numpy as np
+import sys, time, subprocess
+#****************SUBROUTINES*****************
+
+def LSST_DC2_Tree_Ring_Model():
+    # Craig Lage UC Davis 16-Mar-18; cslage@ucdavis.edu
+    # This function returns a tree ring model drawn from an anlytical function that was
+    # derived based on tree ring data collected by Hye-Yun Park at BNL.  The data
+    # used is in imSim/data/tree_ring_data, and a description of the method is in
+    # imSim/data/tree_ring_data/Tree_Rings_13Feb18.pdf
+    # Based on the data, 40% of the sensors are assumed to have 'bad' tree rings, with an
+    # amplitude 10X greater that the 60% of the sensors that have 'good' tree rings.
+
+    bad_tree_ring_fraction = 0.40 # Fraction of sensors that have 'bad' tree rings.
+    A = 2E-3 # Baseline level of pixel deviation
+    numfreqs1 = 15 # Number of spatial frequencies
+    meank1 = 60.0 # Mean spatial freqency in pixels
+    sigmak1 = 10.0 # Standard deviation of spatial freqency in pixels
+    numfreqs2 = 5 # Number of spatial frequencies
+    meank2 = 35.0 # Mean spatial freqency in pixels
+    sigmak2 = 10.0 # Standard deviation of spatial freqency in pixels
+
+    # This generates the parameters for radial tree ring function.
+    # Tree ring amplitude in pixels is given by A + B * r^4
+    # a list of spatial frequencies and phases is generated for each detector
+    if np.random.rand() < bad_tree_ring_fraction:
+        B = np.random.uniform(4.0E-17, 8.0E-17, 1)[0] # Amplitude factor in pixels^-3 
```

Comment:
Should this function accept as an argument an instantiation of `np.random.RandomState`?  That way, we can control the seed of the random number generator and have reproducible results.